node/tests/SharedTests.ts

```diff
@@ -2527,6 +2527,130 @@ export function runBaseTests<Context>(config: {
+
+        const setRes2 = await client.set(key, value, {
+            expiry: {
+                type: "seconds",
+                count: 1,
+            },
+        });
+        expect(setRes2).toEqual("OK");
+        const getRes2 = await client.get(key);
+        expect(getRes2).toEqual(value);
+
+        const setRes3 = await client.set(key, value, {
+            expiry: "keepExisting",
+        });
+        expect(setRes3).toEqual("OK");
+        const getRes3 = await client.get(key);
+        expect(getRes3).toEqual(value);
+
+        const setRes4 = await client.set(key, value, {
+            expiry: {
+                type: "unixMilliseconds",
+                count: 2,
+            },
+        });
+        expect(setRes4).toEqual("OK");
+        // wait for the key to expire
+        setTimeout(() => {}, 5);
+        const getRes4 = await client.get(key);
+        // key should have expired
+        expect(getRes4).toEqual(null);
+    }
+
+    async function setWithExistOptions(
+        client: BaseClient,
+        key: string,
+        value: string,
+    ) {
+        const setRes = await client.set(key, value, {
+            conditionalSet: "onlyIfExists",
+        });
+        expect(setRes).toEqual("OK");
+        const getRes = await client.get(key);
+        expect(getRes).toEqual(value);
+
+        const setRes2 = await client.set(key, value + "1", {
+            conditionalSet: "onlyIfDoesNotExist",
+        });
+        // key already exists, so it should not be set
+        expect(setRes2).toEqual(null);
+        // key was already set, so it should not be changed
+        const getRes2 = await client.get(key);
+        expect(getRes2).toEqual(value);
+    }
+
+    async function setWithGetOldOptions(
+        client: BaseClient,
+        key: string,
+        value: string,
+    ) {
+        const setRes = await client.set(key, value, {
+            returnOldValue: true,
+        });
+        expect(setRes).toEqual(value);
+        const getRes = await client.get(key);
+        expect(getRes).toEqual(value);
+        const setRes2 = await client.set(key + "1", value, {
+            returnOldValue: true,
+        });
+        // key did not exist, so it should not return a value
+        expect(setRes2).toEqual(null);
+        const getRes2 = await client.get(key + "1");
+        expect(getRes2).toEqual(value);
+    }
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "Set commands with options test_%p",
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key = uuidv4();
+                const value = uuidv4();
+                await classicSet(client, key, value);
+                await setWithExpiryOptions(client, key, value);
+                await setWithExistOptions(client, key, value);
+                await setWithGetOldOptions(client, key, value);
+                // try to set with all options, use + 1 to avoid key collision
```

Comment:
you say "avoid key collision" but also set "onlyIfExists" suggesting that you require that the key is set or else you'll get a `null`.  I think, somewhere, in your test, you've set `key + "1"` or else this test would fail, right?